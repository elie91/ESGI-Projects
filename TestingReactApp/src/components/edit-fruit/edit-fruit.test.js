import React from 'react';
import {shallow,} from 'enzyme';
import EditFruit from "./edit-fruit";

describe('EditFruit component', () => {

  let wrapper;
  let mockFruits;
  let mockLocation;
  let mockProps;

  beforeEach(() => {
    mockFruits = [];
    mockLocation = {
      state: {
        fruit: {
          '@id': "/fruits/14",
          '@type': "Fruit",
          cartFruits: [],
          id: 14,
          name: "test"
        }
      }
    }

    mockProps = {
      location: mockLocation,
      fruits: mockFruits
    };

    wrapper = shallow(<EditFruit {...mockProps} />)
  });


  it('should render EditFruit component', () => {
    expect(wrapper).toMatchSnapshot();
  });

});

