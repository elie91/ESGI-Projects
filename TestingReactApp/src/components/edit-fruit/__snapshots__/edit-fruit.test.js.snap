// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFruit component should render EditFruit component 1`] = `ShallowWrapper {}`;
